README.md

```diff
@@ -40,9 +40,40 @@ the values in the cells are the distances between the corresponding cities) and
 returns the length of the shortest tour (not the tour itself).
 
 Test your new function; I've provided some basic testing code in `code.test.js`.
+## Plagarism Statement
+
+All exercises must contain the following statement:
+“I certify that I have listed all sources used to complete this exercise, including the use
+of any Large Language Models. All of the work is my own, except where stated
+otherwise. I am aware that plagiarism carries severe penalties and that if plagiarism is
+suspected, charges may be filed against me without prior notice.”
+
+## Note
+I did not complete this excersise last semester, so I am attempting this completely from scratch. I used this video "Held Karp Dynamic Programming Algorithm for the Traveling Salesman Problem Example" (https://www.youtube.com/watch?v=6jqlBDYNrL0) as an example for my algorithm, the video itself has no code, the way he navigated and explained the problem made sense to me so it helped to see it computed by hand.
+I also used bitmasking in my algorithm but didn't know fully how it operated and the syntax so I asked chatGPT "what is a bitmask and how would it represent cities" and it gave me this verbatum "What Is a Bitmask?
+A bitmask is a way to represent subsets of a set using binary numbers. Each bit in the binary number corresponds to whether an element is included in the subset:
```

Comment:
To be honest, I was up late and trying to work through it, and it was mentioned once in the video and Lily Brongo was discussing possible solutions to a friend Alex Bryan (not in the class) before mobile programming one day and for some reason that's all I had in my head while trying to work through this and I never really got away from that paradigm. Is it a bad implementation then, or did I make it unnecessarily complicated?

Reply:
It's just hard to understand what you're doing, and as far as I can see you can achieve the same with much easier code.

Reply:
I think I agree with you there, as long as it isn't too terrible of an implement (albeit overcomplicated) would it be ok for you to approve this anyway? I looked ahead and I think I would like to find a different implementation to try to use for comparison if I can reach it and get through the other tsp too, I won't use bitmasks again though. I just feel I'm running out of time and would appreciate just the slight leeway here so I can get to work on other exercises please.

Reply:
Well the issue is that according to your comments you didn't write this, but an AI did -- and this is the part that does the dynamic programming. So you haven't implemented a solution.

Reply:
I just try to overcomment when it comes to code references just in case, but I had spent a few hours trying to get that innermost line correct and trying to see at what step my issues were occurring, no matter what I tried it would evaluate just slightly off. It was the tool that eventually suggested the right correction, but I really tried my best to debug on my own first and spent a good deal of time tryingnto fix it. I understand it's still true that the very end result had the ai tool so I'm ok going back and implementing with sets instead, is that what you would like?

Reply:
You'll have to do a different implementation to get the credit for it.

Reply:
I think i also understand using a cache with keys for memorization would that be ok?

Reply:
Any solution to the exercise that is your own work is ok.

Reply:
Ok, I've taken a few extra days to work through a new implementation using parts of what I've learned from some of the newer exercises I've completed, everything I read I cited and my other repos I also used for inspiration are cited too and all the tests are passing again. I think I'm ready for re-review